sympy/series/gruntz.py

```diff
@@ -442,6 +443,18 @@ def limitinf(e, x, leadsimp=False):
         p = Dummy('p', positive=True)
         e = e.subs(x, p)
         x = p
+    if e.has(floor):
```

Comment:
`limitinf` is not the proper place for checking discontinuous functions like `floor`. I would add asymptotic series support to `floor`. (`_eval_aseries`)

Reply:
Alright, but where will `_eval_aseries` for `floor` be called in gruntz? After adding support for `_eval_aseries`, how and where do we incorporate it into the gruntz algo? From what I understand, you are asking to remove the check for `e.has(floor)` from gruntz completely right? Or do you want me to keep the check but call `_eval_aseries` instead?

Reply:
`gruntz` should not need to check specific functions like `floor`. It relies on series expansions, either power series or asymptotic expansions (when there is no power series).